### PR TITLE
feat(parser.prometheusremotewrite): Parse and generate histogram buckets

### LINKED ISSUE #14890: Missing histogram measurements in Prometheus Remote Write Spec
### Relevant telegraf.conf

```toml
[agent]
  interval      = "2s"
  omit_hostname = true
  flush_interval = "4s"

[[inputs.http_listener_v2]]
  service_address = ":1234"
  paths = ["/receive"]
  data_format = "prometheusremotewrite"

[[processors.starlark]]
source = '''
def apply(metric):
  if metric.name == "prometheus_remote_write":
      for k, v in metric.fields.items():
          metric.name = k
          metric.fields["value"] = v
          metric.fields.pop(k)
  return metric
'''

[[outputs.file]]
  files = ["stdout"]
  data_format = "influx"
```

### Logs from Telegraf

```text
2024-02-26T12:35:26Z I! Tags enabled: 
2024-02-26T12:35:26Z I! [agent] Config: Interval:2s, Quiet:false, Hostname:"", Flush Interval:4s
2024-02-26T12:35:26Z D! [agent] Initializing plugins
2024-02-26T12:35:26Z D! [agent] Connecting outputs
2024-02-26T12:35:26Z D! [agent] Attempting connection to [outputs.file]
2024-02-26T12:35:26Z D! [agent] Successfully connected to outputs.file
2024-02-26T12:35:26Z D! [agent] Starting service inputs
2024-02-26T12:35:26Z I! [inputs.http_listener_v2] Listening on [::]:1234
k6_http_reqs_total,expected_response=true,method=GET,name=test,proto=HTTP/1.1,scenario=default,status=200,testrun=testrun-test,url=test,workflow=CustomRun value=100 1708950930462000000
k6_checks_rate,check=GET\ Request,scenario=default,testrun=testrun-test,workflow=CustomRun value=1 1708950930462000000
k6_iterations_total,scenario=default,testrun=testrun-test,workflow=CustomRun value=100 1708950930467000000
k6_data_sent_total,scenario=default,testrun=testrun-test,workflow=CustomRun value=8000 1708950930467000000
k6_data_sent_total,group=::setup,testrun=testrun-test,workflow=CustomRun value=0 1708950929764000000
k6_data_received_total,group=::setup,testrun=testrun-test,workflow=CustomRun value=0 1708950929764000000
k6_http_req_failed_rate,expected_response=true,method=GET,name=test,proto=HTTP/1.1,scenario=default,status=200,testrun=testrun-test,url=test,workflow=CustomRun value=0 1708950930462000000
k6_data_received_total,scenario=default,testrun=testrun-test,workflow=CustomRun value=85300 1708950930467000000
2024-02-26T12:35:30Z D! [outputs.file] Wrote batch of 8 metrics in 2.166625ms
```

### System info

Telegraf 1.28.3, macOS sonoma v14.3.1

### Docker

_No response_

### Steps to reproduce

1. Write any K6 script(scripts) or pick from docs
2. Run the script with native histograms enabled along with telegraf config running parallely,
`K6_PROMETHEUS_RW_SERVER_URL=http://localhost:1234/receive \
K6_PROMETHEUS_RW_TREND_AS_NATIVE_HISTOGRAM=true \
k6 run -o experimental-prometheus-rw script.js -i 100` (for example used 100 iterations)

### Expected behavior

The output should have included all the native histogram metrics as described [here](https://k6.io/docs/results-output/real-time/prometheus-remote-write/)

### Actual behavior

The measurements in the telegraf stdout are only of type counter and rate, to be precise 6 different measurements devoid of histogram measurements. For multiple vus, two more measurements **k6_vus** and **k6_vus_max** are recieved.

### Additional info

Ref. Link for telegraf [config](https://github.com/influxdata/telegraf/tree/master/plugins/parsers/prometheusremotewrite)

### PR DESCRIPTION
**Pull request by powersj**
## Summary

Enable the parsing of histograms that we are currently entirely ignoring in prometheus remote write plugin.

## Checklist

- [x] No AI generated code was used in this PR

## Related issues

resolves #14890

### CODE REVIEW

**Comment by powersj** [comment]
> @powersj is your "Do not merge" tag in the description still valid?

It wasn't, but the user came back with a comment, so now I need to go look into something. I'll remove the waiting for final review tag.

**Comment by powersj** [comment]
ok updated histograms, rebased on master. Read for review.

**Review by srebhan — APPROVED** [review]
Thanks @powersj!

**Review by DStrand1 — APPROVED** [review]

**Pull request merged**
